src/app/portlets/content-types/fields/content-type-fields-row/content-type-fields-row.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, Input } from '@angular/core';
+import { Field, FieldRow } from '../service';
+
+/**
+ * Show all the Field Types
+ *
+ * @export
+ * @class FieldTypesContainerComponent
+ */
+@Component({
+    selector: 'content-type-fields-row',
+    styles: [require('./content-type-fields-row.component.scss')],
+    templateUrl: './content-type-fields-row.component.html',
+})
+export class ContentTypeFieldsRowComponent {
+    @Input() fieldRow: FieldRow;
+
+    removeField(field: Field): void {
+        this.fieldRow.columns = this.fieldRow.columns.map(col => {
+            let index: number = col.fields.indexOf(field);
+
+            if (index !== -1) {
+                col.fields.splice(index, 1);
+             }
+            return col;
+        });
+    }
+
+    getSpanWidth(): string {
+        let nColumns = this.fieldRow.columns.length;
+        return `${(100 - nColumns) / nColumns}%`;
```

Comment:
Shouldn't this be ```100 / nColumns```? If your having problems with the positioning this is something css can fix, let me know.